scripts/error-extractor/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+var path = require('path');
+var traverse = require('babel-traverse').default;
+
+var evalToString = require('./evalToString');
+var getFilePaths = require('./getFilePaths');
+var invertObject = require('./invertObject');
+var parseJSFile = require('./parseJSFile');
+var writeJSON = require('./writeJSON');
+
+var outputFileName = 'ErrorMap.json';
+var outputFilePath = path.join(__dirname, '../../', outputFileName);
+
+var existingErrorMap;
+var currentID;
+
+try {
+  existingErrorMap = require(outputFilePath);
+} catch (e) {
+  existingErrorMap = {};
+}
+
+// Now the error map is in the format of `{ 0: 'MUCH ERROR', 1: 'SUCH WRONG' }`
+
+var currentIDNumbers = Object.keys(existingErrorMap).map(function(str) {
+  return parseInt(str, 10);
```

Comment:
nit: `+str` is all you need here; I tend to avoid `parseInt` because rarely do you want the behavior that `parseInt('2a')` gives `2`. You could also use `Number(str)`. Also, Math.max casts to number so you probably don't even need this at all. Maybe all you need is

```
var currentID = Math.max.apply(null, [-1].concat(Object.keys(existingErrorMap)));
```

though the `-1` thing is admittedly a little cryptic and your if statement is probably clearer.